File: app/src/main/java/com/electricdreams/shellshock/feature/settings/CurrencySettingsActivity.java
package com.electricdreams.shellshock.feature.settings;

import android.os.Bundle;
import android.view.View;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import androidx.appcompat.app.AppCompatActivity;

import com.electricdreams.shellshock.R;
import com.electricdreams.shellshock.core.util.CurrencyManager;

public class CurrencySettingsActivity extends AppCompatActivity {
    
    private RadioGroup currencyRadioGroup;
    private RadioButton radioUsd;
    private RadioButton radioEur;
    private RadioButton radioGbp;
    private RadioButton radioJpy;
    private CurrencyManager currencyManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_currency_settings);
        
        // Setup back button
        View backButton = findViewById(R.id.back_button);
        if (backButton != null) {
            backButton.setOnClickListener(v -> finish());
        }
        
        // Initialize manager
        currencyManager = CurrencyManager.getInstance(this);
        
        // Initialize views
        currencyRadioGroup = findViewById(R.id.currency_radio_group);
        radioUsd = findViewById(R.id.radio_usd);
        radioEur = findViewById(R.id.radio_eur);
        radioGbp = findViewById(R.id.radio_gbp);
        radioJpy = findViewById(R.id.radio_jpy);
        
        // Set current selection
        setSelectedCurrency(currencyManager.getCurrentCurrency());
        
        // Auto-save on selection change
        currencyRadioGroup.setOnCheckedChangeListener((group, checkedId) -> {
            String selectedCurrency = getSelectedCurrency();
            currencyManager.setPreferredCurrency(selectedCurrency);
        });
    }
    
    private void setSelectedCurrency(String currencyCode) {
        switch (currencyCode) {
            case CurrencyManager.CURRENCY_EUR:
                radioEur.setChecked(true);
                break;
            case CurrencyManager.CURRENCY_GBP:
                radioGbp.setChecked(true);
                break;
            case CurrencyManager.CURRENCY_JPY:
                radioJpy.setChecked(true);
                break;
            case CurrencyManager.CURRENCY_USD:
            default:
                radioUsd.setChecked(true);
                break;
        }
    }
    
    private String getSelectedCurrency() {
        int selectedId = currencyRadioGroup.getCheckedRadioButtonId();
        
        if (selectedId == R.id.radio_eur) {
            return CurrencyManager.CURRENCY_EUR;
        } else if (selectedId == R.id.radio_gbp) {
            return CurrencyManager.CURRENCY_GBP;
        } else if (selectedId == R.id.radio_jpy) {
            return CurrencyManager.CURRENCY_JPY;
        } else {
            return CurrencyManager.CURRENCY_USD;
        }
    }
}

File: app/src/main/java/com/electricdreams/numo/feature/settings/CurrencySettingsActivity.kt
package com.electricdreams.numo.feature.settings

import android.os.Bundle
import android.view.View
import android.widget.RadioButton
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity
import com.electricdreams.numo.R
import com.electricdreams.numo.core.util.CurrencyManager

class CurrencySettingsActivity : AppCompatActivity() {

    private lateinit var currencyRadioGroup: RadioGroup
    private lateinit var radioUsd: RadioButton
    private lateinit var radioEur: RadioButton
    private lateinit var radioGbp: RadioButton
    private lateinit var radioJpy: RadioButton
    private lateinit var currencyManager: CurrencyManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_currency_settings)

        findViewById<View?>(R.id.back_button)?.setOnClickListener { finish() }

        currencyManager = CurrencyManager.getInstance(this)

        currencyRadioGroup = findViewById(R.id.currency_radio_group)
        radioUsd = findViewById(R.id.radio_usd)
        radioEur = findViewById(R.id.radio_eur)
        radioGbp = findViewById(R.id.radio_gbp)
        radioJpy = findViewById(R.id.radio_jpy)

        // Use the persisted preferred currency for initial selection.
        // NOTE: Items in the editable catalog are normalized to this
        // currency on load (see ItemManager). We intentionally do **not**
        // rewrite historical data like receipts or payment history so that
        // past checkouts still reflect the original currency they were
        // created in.
        //
        // Active flows (item prices, baskets, checkout) always use this
        // single preferred fiat currency.
        setSelectedCurrency(currencyManager.getCurrentCurrency())

        currencyRadioGroup.setOnCheckedChangeListener { _, _ ->
            val selectedCurrency = getSelectedCurrency()
            currencyManager.setPreferredCurrency(selectedCurrency)
        }
    }

    private fun setSelectedCurrency(currencyCode: String) {
        when (currencyCode) {
            CurrencyManager.CURRENCY_EUR -> radioEur.isChecked = true
            CurrencyManager.CURRENCY_GBP -> radioGbp.isChecked = true
            CurrencyManager.CURRENCY_JPY -> radioJpy.isChecked = true
            CurrencyManager.CURRENCY_USD -> radioUsd.isChecked = true
            else -> radioUsd.isChecked = true
        }
    }

    private fun getSelectedCurrency(): String {
        val selectedId = currencyRadioGroup.checkedRadioButtonId
        return when (selectedId) {
            R.id.radio_eur -> CurrencyManager.CURRENCY_EUR
            R.id.radio_gbp -> CurrencyManager.CURRENCY_GBP
            R.id.radio_jpy -> CurrencyManager.CURRENCY_JPY
            else -> CurrencyManager.CURRENCY_USD
        }
    }
}
